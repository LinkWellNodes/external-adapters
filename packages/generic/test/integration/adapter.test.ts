import { authorizationApiKeyTests } from './endpoint-generic-request/authorization-api-key'
import { authorizationBasicAuthTests } from './endpoint-generic-request/authorization-basic-auth'
import { authorizationBearerTokenTests } from './endpoint-generic-request/authorization-bearer-token'
import { authorizationCustomTests } from './endpoint-generic-request/authorization-custom'
import { authorizationNoAuthTests } from './endpoint-generic-request/authorization-no-auth'

// NB: run sequentially due to incapacity of assigning an independent port
// (e.g. 8080, 8081 via process.env.EA_PORT) per test server
describe('endpoint: generic-request', () => {
  describe('authorization type: api_key', () => authorizationApiKeyTests())
  describe('authorization type: basic_auth', () => authorizationBasicAuthTests())
  describe('authorization type: bearer_token', () => authorizationBearerTokenTests())
  describe('authorization type: custom', () => authorizationCustomTests())
  describe('authorization type: no_auth', () => authorizationNoAuthTests())
})
