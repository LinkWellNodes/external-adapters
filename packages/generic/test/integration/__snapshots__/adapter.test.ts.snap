// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`endpoint: generic-request authorization type: api_key success calls returns the expected result, when credentials location is 'headers', and API 'response.data' is a JSON object 1`] = `
Object {
  "data": Object {
    "result": Object {
      "value": 42.777,
    },
    "value": 42.777,
  },
  "jobRunID": "1",
  "result": Object {
    "value": 42.777,
  },
  "statusCode": 200,
}
`;

exports[`endpoint: generic-request authorization type: api_key success calls returns the expected result, when credentials location is 'headers', and API 'response.data' is not a JSON object 1`] = `
Object {
  "data": Object {
    "data": Array [
      Object {
        "value": 42.777,
      },
    ],
    "result": Array [
      Object {
        "value": 42.777,
      },
    ],
  },
  "jobRunID": "1",
  "result": Array [
    Object {
      "value": 42.777,
    },
  ],
  "statusCode": 200,
}
`;

exports[`endpoint: generic-request authorization type: api_key success calls returns the expected result, when credentials location is 'params', and API 'response.data' is a JSON object 1`] = `
Object {
  "data": Object {
    "result": Object {
      "value": 42.777,
    },
    "value": 42.777,
  },
  "jobRunID": "1",
  "result": Object {
    "value": 42.777,
  },
  "statusCode": 200,
}
`;

exports[`endpoint: generic-request authorization type: api_key success calls returns the expected result, when credentials location is 'params', and API 'response.data' is not a JSON object 1`] = `
Object {
  "data": Object {
    "data": Array [
      Object {
        "value": 42.777,
      },
    ],
    "result": Array [
      Object {
        "value": 42.777,
      },
    ],
  },
  "jobRunID": "1",
  "result": Array [
    Object {
      "value": 42.777,
    },
  ],
  "statusCode": 200,
}
`;

exports[`endpoint: generic-request authorization type: basic_auth success calls returns the expected result, when API 'response.data' is a JSON object 1`] = `
Object {
  "data": Object {
    "result": Object {
      "value": 42.777,
    },
    "value": 42.777,
  },
  "jobRunID": "1",
  "result": Object {
    "value": 42.777,
  },
  "statusCode": 200,
}
`;

exports[`endpoint: generic-request authorization type: basic_auth success calls returns the expected result, when API 'response.data' is not a JSON object 1`] = `
Object {
  "data": Object {
    "data": Array [
      Object {
        "value": 42.777,
      },
    ],
    "result": Array [
      Object {
        "value": 42.777,
      },
    ],
  },
  "jobRunID": "1",
  "result": Array [
    Object {
      "value": 42.777,
    },
  ],
  "statusCode": 200,
}
`;

exports[`endpoint: generic-request authorization type: bearer_token success calls returns the expected result, when API 'response.data' is a JSON object 1`] = `
Object {
  "data": Object {
    "result": Object {
      "value": 42.777,
    },
    "value": 42.777,
  },
  "jobRunID": "1",
  "result": Object {
    "value": 42.777,
  },
  "statusCode": 200,
}
`;

exports[`endpoint: generic-request authorization type: bearer_token success calls returns the expected result, when API 'response.data' is not a JSON object 1`] = `
Object {
  "data": Object {
    "data": Array [
      Object {
        "value": 42.777,
      },
    ],
    "result": Array [
      Object {
        "value": 42.777,
      },
    ],
  },
  "jobRunID": "1",
  "result": Array [
    Object {
      "value": 42.777,
    },
  ],
  "statusCode": 200,
}
`;

exports[`endpoint: generic-request authorization type: custom success calls returns the expected result, when API 'response.data' is a JSON object 1`] = `
Object {
  "data": Object {
    "result": Object {
      "value": 42.777,
    },
    "value": 42.777,
  },
  "jobRunID": "1",
  "result": Object {
    "value": 42.777,
  },
  "statusCode": 200,
}
`;

exports[`endpoint: generic-request authorization type: custom success calls returns the expected result, when API 'response.data' is not a JSON object 1`] = `
Object {
  "data": Object {
    "data": Array [
      Object {
        "value": 42.777,
      },
    ],
    "result": Array [
      Object {
        "value": 42.777,
      },
    ],
  },
  "jobRunID": "1",
  "result": Array [
    Object {
      "value": 42.777,
    },
  ],
  "statusCode": 200,
}
`;

exports[`endpoint: generic-request authorization type: no_auth success calls returns the expected result, when API 'response.data' is a JSON object 1`] = `
Object {
  "data": Object {
    "result": Object {
      "value": 42.777,
    },
    "value": 42.777,
  },
  "jobRunID": "1",
  "result": Object {
    "value": 42.777,
  },
  "statusCode": 200,
}
`;

exports[`endpoint: generic-request authorization type: no_auth success calls returns the expected result, when API 'response.data' is not a JSON object 1`] = `
Object {
  "data": Object {
    "data": Array [
      Object {
        "value": 42.777,
      },
    ],
    "result": Array [
      Object {
        "value": 42.777,
      },
    ],
  },
  "jobRunID": "1",
  "result": Array [
    Object {
      "value": 42.777,
    },
  ],
  "statusCode": 200,
}
`;
